### PR TITLE
Adding timeline for crop cycle

### PR DESCRIPTION
**Pull request by gautamguptabasant**
Screenshot 2021-01-31 at 8 29 12 PM

- Renders a timeline for the selected crop cycle based 
- The timeline included crop cycle processes and inspections associated with the crop cycle

### CODE REVIEW

File: src/Pages/CropCycleTimeline.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  },
+  formControl: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    minWidth: 300,
+    maxWidth: 300,
+  },
+  submitButton: {
```
Comment (mahimavuppuluri) [inline src/Pages/CropCycleTimeline.js:33]: remove styles not being used

File: src/Pages/CropCycleTimeline.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  const [alertSeverity, setAlertSeverity] = useState('');
+
+  const handleAlertClose = (_event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setAlertStatus(false);
+  };
+
+  const showAlert = (message, severity) => {
```
Comment (aashnaPrakash) [inline src/Pages/CropCycleTimeline.js:74]: function documentation?

File: src/Pages/CropCycleTimeline.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+  /**
+   * Formats a crop cycle object for display by using the crop name,
+   * season and year separated by hyphens for better readability
+   * @param {object} cropCycleObject Object containing crop cycle details
+   */
+  const formatCropCycleForDisplay = (cropCycleObject) =>
+    `${cropCycleObject.crop.name} - ${cropCycleObject.season} - ${cropCycleObject.year}`;
+
+  const handleChange = ({ target }) => {
```
Comment (aashnaPrakash) [inline src/Pages/CropCycleTimeline.js:178]: function documentation?

File: src/components/common/AmsTimeline.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+ * Renders a timeline component that includes all the processes and
+ * inspections associated with a crop cycle.
+ *
+ * @param {Array} processes All the crop cycles processes
+ * @param {Array} inspections All the inspections recorded on the crop cycle
+ */
+const AmsTimeline = ({ processes, inspections }) => {
+  const classes = useStyles();
+
+  const mergeProcessesAndInspections = () => {
```
Comment (aashnaPrakash) [inline src/components/common/AmsTimeline.js:53]: comment stating what the function is used for?

**Review by aashnaPrakash — APPROVED** [review]

File: src/components/common/AmsTimeline.js (hunk 4 above)
Comment (gautamguptabasant) [reply src/components/common/AmsTimeline.js:53]: Pretty trivial function and the name should give good enough indication of what is happening.

File: src/Pages/CropCycleTimeline.js (hunk 3 above)
Comment (gautamguptabasant) [reply src/Pages/CropCycleTimeline.js:178]: Trivial method

File: src/Pages/CropCycleTimeline.js (hunk 2 above)
Comment (gautamguptabasant) [reply src/Pages/CropCycleTimeline.js:74]: Common method used across all the pages, trivial.

File: src/Pages/CropCycleTimeline.js (hunk 1 above)
Comment (gautamguptabasant) [reply src/Pages/CropCycleTimeline.js:33]: Let me update it in the next PR.

**Pull request merged**